agent/engine/engine_unix_integ_test.go

```diff
@@ -296,6 +297,390 @@ func TestStartStopUnpulledImageDigest(t *testing.T) {
 	VerifyTaskStoppedStateChange(t, taskEngine)
 }
 
+// Tests that containers with ordering dependencies are able to reach MANIFEST_PULLED state
+// regardless of the dependencies.
+func TestManifestPulledDoesNotDependOnContainerOrdering(t *testing.T) {
+	imagePullBehaviors := []config.ImagePullBehaviorType{
+		config.ImagePullDefaultBehavior, config.ImagePullAlwaysBehavior,
+		config.ImagePullPreferCachedBehavior, config.ImagePullOnceBehavior,
+	}
+
+	for _, behavior := range imagePullBehaviors {
+		t.Run(fmt.Sprintf("%v", behavior), func(t *testing.T) {
+			cfg := DefaultTestConfigIntegTest()
+			cfg.ImagePullBehavior = behavior
+			cfg.DockerStopTimeout = 100 * time.Millisecond
+			taskEngine, done, _, _ := SetupIntegTestTaskEngine(cfg, nil, t)
+			defer done()
+
+			first := createTestContainerWithImageAndName(testRegistryImage, "first")
```

Comment:
If we use a Windows image from public ECR, is it trivial to support this test on Windows?

Reply:
I've changed `testRegistryImage` to point the Windows Server image from MCR in https://github.com/aws/amazon-ecs-agent/pull/4304/commits/725e2b785003dc099eea5e25cc9fa21b9e10c9f3. (We're already using it in `TestStartStopUnpulledImage` and `TestStartStopUnpulledImageDigest`.)

Reply:
I've changed `testRegistryImage` to point the Windows Server image from MCR in https://github.com/aws/amazon-ecs-agent/pull/4304/commits/725e2b785003dc099eea5e25cc9fa21b9e10c9f3. (We're already using it in `TestStartStopUnpulledImage` and `TestStartStopUnpulledImageDigest`.)

Reply:
Thanks! So do we still need to move `TestManifestPulledDoesNotDependOnContainerOrdering` and other tests to this unix specific test file?